### PR TITLE
Add remaining account manager methods to message interface

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add remaining account manager methods to message interface

## Links to any relevant issues

https://github.com/iotaledger/wallet.rs/issues/1531

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/message_interface/message_handler.rs [hunk 1]
```diff
@@ -330,6 +330,35 @@ impl WalletMessageHandler {
...
                 .await
             }
+            #[cfg(feature = "participation")]
+            Message::GetParticipationEvent(event_id) => {
+                convert_async_panics(|| async {
+                    let event = self
+                        .account_manager
+                        .get_participation_event(event_id)
+                        .await?
+                        .ok_or_else(|| crate::Error::Storage(format!("event {event_id} not found")))?
```
Comment (thibault-martinez) [inline src/message_interface/message_handler.rs]: Feels a bit odd to return an error, I would put an option in the Response, we do that for some other types

File: src/message_interface/message_handler.rs [hunk 2]
```diff
@@ -330,6 +330,34 @@ impl WalletMessageHandler {
...
+                        .await?
+                        .map(|(e, _)| e);
+                    Ok(Response::ParticipationEvent(event))
+                })
+                .await
+            }
+            #[cfg(feature = "participation")]
+            Message::GetParticipationEventStatus(event_id) => {
+                convert_async_panics(|| async {
+                    let event_status = self.account_manager.get_participation_event_status(event_id).await?;
```
Comment (thibault-martinez) [inline src/message_interface/message_handler.rs:348]: I know I previously approved this but I also think this should return Option and not an error if the record is not found

Comment (Thoralf-M) [reply src/message_interface/message_handler.rs:348]: But this is the event status which gets requested from the node

Comment (thibault-martinez) [reply src/message_interface/message_handler.rs:348]: Not sure what you mean. What happens if you pass a bullshit event ID?

Comment (Thoralf-M) [reply src/message_interface/message_handler.rs:348]: Then it should return an error, that the event wasn't found and not None

Comment (thibault-martinez) [reply src/message_interface/message_handler.rs:348]: Why does `get_participation_event` return an option and not `get_participation_event_status`? They are doing pretty similar stuff ?

Comment (Thoralf-M) [reply src/message_interface/message_handler.rs:348]: get_participation_event works with local data from the db, get_participation_event_status returns what the node returns for this event

Comment (thibault-martinez) [reply src/message_interface/message_handler.rs:348]: Ok I see. I now remember asking in the past why we don't return Option for the routes, this still kinda puzzles me a bit. But fine 👍🏻

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**